src/watershed.jl

```diff
@@ -39,22 +51,61 @@ function watershed(img::AbstractArray{T, N}, markers::AbstractArray{S,N}) where
             for j in CartesianIndices(_colon(max(Istart,i-one(i)), min(i+one(i),Iend)))
                 if segments[j] == 0
                     segments[j] = markers[i]
-                    enqueue!(pq, j, PixelKey(img[i], time_step))
+                    enqueue!(pq, j, PixelKey(img[i], time_step, j))
                     time_step += 1
                 end
             end
         end
     end
 
     while !isempty(pq)
-        current = dequeue!(pq)
-        segments_current = segments[current]
-        img_current = img[current]
-        for j in CartesianIndices(_colon(max(Istart,current-one(current)), min(current+one(current),Iend)))
+        curr_idx, curr_elem = dequeue_pair!(pq)
+        segments_current = segments[curr_idx]
+
+        # If we're using the compact algorithm, we need assign grouping for a given location
+        # when it comes off the queue since we could have found a better suited watershed later.
```

Comment:
```suggestion
        # when it comes off the queue since we could find a better suited watershed later.
```
(just making the tenses consistent, if this isn't the right change ignore it)